### PR TITLE
LPS-212337

### PR DESCRIPTION
**Pull request by younes2123**

### CODE REVIEW

**Comment by pedroamarall** [comment]
ci:test:sf

**Comment by pedroamarall** [comment]
Hello @younes2123, you already have an [open pr](https://github.com/pedroamarall/liferay-portal/pull/14) with the same files, could you put these changes in the pr that is open please

**Comment by younes2123** [comment]
> Hello @younes2123, you already have an [open pr](https://github.com/pedroamarall/liferay-portal/pull/14) with the same files, could you put these changes in the pr that is open please

I closed the old one.

File: portal-web/test/functional/com/liferay/portalweb/tests/mytests/paths/PanelTest.path [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+<body>
+<table border="1" cellpadding="1" cellspacing="1">
+<thead>
+<tr><td rowspan="1" colspan="3">PanelTest</td></tr>
+</thead>
+
+<tbody>
+<tr>
+	<td>MENU_BUTTON</td>
+	<td>//*[contains(@class, 'dropdown-toggle lfr-portal-tooltip btn btn-monospaced btn-sm btn-unstyled')]</td>
```
Comment (manoelcyreno) [inline portal-web/test/functional/com/liferay/portalweb/tests/mytests/paths/PanelTest.path:15]: This locator is a bad practice, cuz you are saying that the classes need to be defined in that order .. if the programmer changes the order your locator will fail, so please, use a better relative XPath to it.

File: portal-web/test/functional/com/liferay/portalweb/tests/mytests/paths/PanelTest.path [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+	<td></td>
+</tr>
+<tr>
+	<td>CONTROL_PANEL_BUTTON</td>
+	<td>//li[contains(@id, '_com_liferay_product_navigation_applications_menu_web_internal_portlet_ProductNavigationApplicationsMenuPortlet_tab_2')]</td>
```
Comment (manoelcyreno) [inline portal-web/test/functional/com/liferay/portalweb/tests/mytests/paths/PanelTest.path:20]: why are you defining the "li" here?

File: portal-web/test/functional/com/liferay/portalweb/tests/mytests/paths/PanelTest.path [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+	<td></td>
+</tr>
+<tr>
+	<td>CONTROL_PANEL_CONFIG_SECTION</td>
+	<td>//h2[contains(@id, 'nav_control_panel.configuration')]</td>
```
Comment (manoelcyreno) [inline portal-web/test/functional/com/liferay/portalweb/tests/mytests/paths/PanelTest.path:25]: why are you defining the "h2" here?

File: portal-web/test/functional/com/liferay/portalweb/tests/mytests/paths/PanelTest.path [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+	<td></td>
+</tr>
+<tr>
+	<td>SIMULATION_MENU_BUTTON</td>
+	<td>//*[@id="_com_liferay_product_navigation_simulation_web_portlet_SimulationPortlet_simulationToggleId"]</td>
+	<td></td>
+</tr>
+<tr>
+	<td>CUSTOM_BUTTON</td>
+	<td>//*[@id="_com_liferay_product_navigation_simulation_web_portlet_SimulationPortlet_simulationDeviceContainer"]/div/div/button[5]</td>
```
Comment (manoelcyreno) [inline portal-web/test/functional/com/liferay/portalweb/tests/mytests/paths/PanelTest.path:35]: This locator is using a mix between an absolute and relative XPath, please, change it to use the **relative xpath** in the right way

File: portal-web/test/functional/com/liferay/portalweb/tests/mytests/macros/MyBlogs.macro [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+		var entryContent = "Blogs Entry Content";
+		var entryTitle = "Blogs Entry Title";
+
+		BlogsNavigator.openBlogsAdmin(siteURLKey = "guest");
+
+		BlogsNavigator.gotoAddEntry();
+
+		BlogsEntry.addEntryContent(
+			entryContent = ${entryContent},
+			entryTitle = ${entryTitle});
```
Comment (manoelcyreno) [inline portal-web/test/functional/com/liferay/portalweb/tests/mytests/macros/MyBlogs.macro:13]: change it to a parameter that the user will pass on the testcase file.

File: portal-web/test/functional/com/liferay/portalweb/tests/mytests/macros/MyBlogs.macro [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+
+		PortletEntry.publish();
+	}
+
+	macro deleteBlogEntry {
+		LexiconEntry.gotoEntryMenuItem(
+			menuItem = "Delete",
+			rowEntry = "Blogs Entry Title");
```
Comment (manoelcyreno) [inline portal-web/test/functional/com/liferay/portalweb/tests/mytests/macros/MyBlogs.macro:21]: change it to a parameter that the user will pass on the testcase file.

File: portal-web/test/functional/com/liferay/portalweb/tests/mytests/macros/MyBlogs.macro [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+	}
+
+	macro editBlogEntry {
+		LexiconEntry.gotoEntryMenuItem(
+			menuItem = "Edit",
+			rowEntry = "Blogs Entry Title");
```
Comment (manoelcyreno) [inline portal-web/test/functional/com/liferay/portalweb/tests/mytests/macros/MyBlogs.macro:27]: change it to a parameter that the user will pass on the testcase file.

File: portal-web/test/functional/com/liferay/portalweb/tests/mytests/macros/MyBlogs.macro [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+
+		BlogsEntry.editTitle(
+			entryTitle = "Blogs Entry Title",
+			entryTitleEdit = "New title");
```
Comment (manoelcyreno) [inline portal-web/test/functional/com/liferay/portalweb/tests/mytests/macros/MyBlogs.macro:31]: change it to a parameter that the user will pass on the testcase file.

File: portal-web/test/functional/com/liferay/portalweb/tests/mytests/macros/MyBlogs.macro [hunk 9]
```diff
@@ -0,0 +1,42 @@
...
+			rowEntry = "Blogs Entry Title");
+	}
+
+	macro editBlogEntry {
+		LexiconEntry.gotoEntryMenuItem(
+			menuItem = "Edit",
+			rowEntry = "Blogs Entry Title");
+
+		BlogsEntry.editTitle(
+			entryTitle = "Blogs Entry Title",
```
Comment (manoelcyreno) [inline portal-web/test/functional/com/liferay/portalweb/tests/mytests/macros/MyBlogs.macro:30]: change it to a parameter that the user will pass on the testcase file.

File: portal-web/test/functional/com/liferay/portalweb/tests/mytests/macros/MyBlogs.macro [hunk 10]
```diff
@@ -0,0 +1,42 @@
...
+			entryTitleEdit = "New title");
+
+		Button.clickPublish();
+	}
+
+	macro searchBlogEntry {
+		BlogsEntry.searchCP(
+			entrySearchTerm = "Blogs",
+			entryTitle = "Blogs Entry Title");
```
Comment (manoelcyreno) [inline portal-web/test/functional/com/liferay/portalweb/tests/mytests/macros/MyBlogs.macro:39]: change it to a parameter that the user will pass on the testcase file.

File: portal-web/test/functional/com/liferay/portalweb/tests/mytests/testcases/MyBlogsTests.testcase [hunk 11]
```diff
@@ -0,0 +1,64 @@
...
+		if (${testPortalInstance} == "true") {
+			PortalInstances.tearDownCP();
+		}
+	}
+
+	@description = "Can Create a Blog"
+	@priority = 5
+	test CanCreateBlog {
+		MyBlogs.addBlogEntry();
+	}
```
Comment (manoelcyreno) [inline portal-web/test/functional/com/liferay/portalweb/tests/mytests/testcases/MyBlogsTests.testcase:26]: where are the "assert" to validate if it is right or not?

File: portal-web/test/functional/com/liferay/portalweb/tests/mytests/testcases/MyBlogsTests.testcase [hunk 12]
```diff
@@ -0,0 +1,64 @@
...
+	@priority = 5
+	test CanDeleteBlog {
+		task ("Given the blog already created") {
+			MyBlogs.addBlogEntry();
+		}
+
+		task ("Then delete the blog entry") {
+			MyBlogs.deleteBlogEntry();
+		}
+	}
```
Comment (manoelcyreno) [inline portal-web/test/functional/com/liferay/portalweb/tests/mytests/testcases/MyBlogsTests.testcase:38]: where are the "assert" to validate if it is right or not?

File: portal-web/test/functional/com/liferay/portalweb/tests/mytests/testcases/MyBlogsTests.testcase [hunk 13]
```diff
@@ -0,0 +1,64 @@
...
+	@priority = 5
+	test CanEditBlog {
+		task ("Given the blog already created") {
+			MyBlogs.addBlogEntry();
+		}
+
+		task ("Then edit the blog title") {
+			MyBlogs.editBlogEntry();
+		}
+	}
```
Comment (manoelcyreno) [inline portal-web/test/functional/com/liferay/portalweb/tests/mytests/testcases/MyBlogsTests.testcase:50]: where are the "assert" to validate if it is right or not?

File: portal-web/test/functional/com/liferay/portalweb/tests/mytests/testcases/MyBlogsTests.testcase [hunk 14]
```diff
@@ -0,0 +1,64 @@
...
+	@priority = 5
+	test CanSearchBlogs {
+		task ("Given the blog already created") {
+			MyBlogs.addBlogEntry();
+		}
+
+		task ("Then search for the blog entry and validate its existensse") {
+			MyBlogs.searchBlogEntry();
+		}
+	}
```
Comment (manoelcyreno) [inline portal-web/test/functional/com/liferay/portalweb/tests/mytests/testcases/MyBlogsTests.testcase:62]: where are the "assert" to validate if it is right or not?

File: portal-web/test/functional/com/liferay/portalweb/tests/mytests/testcases/MyTestFile.testcase [hunk 15]
```diff
@@ -0,0 +1,56 @@
...
+			value1 = 1080);
+
+		TripleClick(locator1 = "PanelTest#HEIGHT_INPUT");
+
+		Type.sendKeys(
+			locator1 = "PanelTest#HEIGHT_INPUT",
+			value1 = 1920);
+
+		Click(locator1 = "PanelTest#APPLY_CUSTOM_BUTTON");
+	}
```
Comment (manoelcyreno) [inline portal-web/test/functional/com/liferay/portalweb/tests/mytests/testcases/MyTestFile.testcase:40]: where are the tasks?

File: portal-web/test/functional/com/liferay/portalweb/tests/mytests/testcases/NewForms.testcase [hunk 16]
```diff
@@ -0,0 +1,126 @@
...
+
+		ProductMenu.gotoPortlet(
+			category = "Content & Data",
+			portlet = "Forms");
+
+		FormsAdminNavigator.gotoViewEntries();
+
+		LexiconEntry.gotoEntryMenuItem(
+			menuItem = "View",
+			rowEntry = "Younes Fatihi");
```
Comment (manoelcyreno) [inline portal-web/test/functional/com/liferay/portalweb/tests/mytests/testcases/NewForms.testcase:123]: refactor all that test to use JSON calls, the test (by UI) needs to do, exactly what you want to validate, not the "setup" of your test.
add tasks to your code to be understandable ....

**Review by manoelcyreno — CHANGES_REQUESTED** [review]
@younes2123 , take a look at:
1- The name of your methods, and the "Then", do not pass to me what in fact it wants to validate... so, please, change it.
2- You didn't add any assert
3- you didn't add tasks in a lot of cases
4- Review all the comments above.

File: portal-web/test/functional/com/liferay/portalweb/tests/mytests/testcases/MyBlogsTests.testcase (hunk 13 above)
Comment (younes2123) [reply portal-web/test/functional/com/liferay/portalweb/tests/mytests/testcases/MyBlogsTests.testcase]: Fixed, please check the last commits

File: portal-web/test/functional/com/liferay/portalweb/tests/mytests/testcases/MyTestFile.testcase (hunk 15 above)
Comment (younes2123) [reply portal-web/test/functional/com/liferay/portalweb/tests/mytests/testcases/MyTestFile.testcase]: Fixed, please check the last commits

File: portal-web/test/functional/com/liferay/portalweb/tests/mytests/testcases/NewForms.testcase (hunk 16 above)
Comment (younes2123) [reply portal-web/test/functional/com/liferay/portalweb/tests/mytests/testcases/NewForms.testcase:123]: I tried to refactor to set the form using JSON calls, but I didn't find any macro or testcase using it, so can you please help me find the JSON macro for forms?

Comment (manoelcyreno) [reply portal-web/test/functional/com/liferay/portalweb/tests/mytests/testcases/NewForms.testcase:123]: @younes2123 , you need to validate what you need to test, so, for example, you are creating the Form, and after you are editing, so, the creation you don't need to create by the UI, you can do it using the file JSONBlog, and the method about JSONBlog.addEntry ...

**Comment by younes2123** [comment]
@manoelcyreno  I refactored the form test by adding tasks but didn't use JSON to set up the form because I think there is no way to do that.

File: portal-web/test/functional/com/liferay/portalweb/tests/mytests/macros/MyBlogs.macro [hunk 17]
```diff
@@ -0,0 +1,39 @@
...
+
+		Button.clickPublish();
+	}
+
+	macro searchBlogEntry {
+		BlogsEntry.searchCP(
+			entrySearchTerm = ${entrySearchTerm},
+			entryTitle = ${entrySearchTerm});
+	}
+
```
Comment (Youness-ELMO) [inline portal-web/test/functional/com/liferay/portalweb/tests/mytests/macros/MyBlogs.macro:38]: this macro is not necessary, you can add this code directly to the testcase file

File: portal-web/test/functional/com/liferay/portalweb/tests/mytests/macros/MyBlogs.macro [hunk 18]
```diff
@@ -0,0 +1,39 @@
...
+			entryTitle = ${entryTitle});
+
+		PortletEntry.publish();
+	}
+
+	macro deleteBlogEntry {
+		LexiconEntry.gotoEntryMenuItem(
+			menuItem = "Delete",
+			rowEntry = ${rowEntry});
+	}
```
Comment (Youness-ELMO) [inline portal-web/test/functional/com/liferay/portalweb/tests/mytests/macros/MyBlogs.macro:19]: no need to create an add macro instead macro u can add it directly

**Comment by younes2123** [comment]
@Youness-ELMO @manoelcyreno  I pushed the refactored code, feel free to review it. 

Thanks.

File: portal-web/test/functional/com/liferay/portalweb/tests/mytests/testcases/MyBlogsTests.testcase [hunk 19]
```diff
@@ -0,0 +1,110 @@
...
+				entryContent = "Blogs Entry Content",
+				entryTitle = "Blogs Entry Title");
+		}
+
+		task ("Then validate the blog is found") {
+			AssertElementPresent(
+				key_entryTitle = "Blogs Entry Title",
+				key_tagName = "Blogs Entry Title",
+				locator1 = "BlogsEntry#TAG");
+		}
```
Comment (manoelcyreno) [inline portal-web/test/functional/com/liferay/portalweb/tests/mytests/testcases/MyBlogsTests.testcase:36]: when?

**Review by manoelcyreno — APPROVED** [review]

**Comment by manoelcyreno** [comment]
looks good

**Comment by manoelcyreno** [comment]
ci:close

**Pull request closed without merging**